src/openrct2/scripting/ScEntity.hpp

```diff
@@ -0,0 +1,177 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2018 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#pragma once
+
+#ifdef __ENABLE_SCRIPTING__
+
+#    include "../common.h"
+#    include "../world/Sprite.h"
+#    include "Duktape.hpp"
+#    include "ScriptEngine.h"
+
+namespace OpenRCT2::Scripting
+{
+    class ScEntity
+    {
+    protected:
+        uint16_t _id = SPRITE_INDEX_NULL;
+
+    public:
+        ScEntity(uint16_t id)
```

Comment:
This class is using snake_case.

Also, the file and class names would be a lot more meaningful as `ScriptEntity` than `ScEntity`